app/src/main/java/ru/uniquenature/myapplication/data/MoviesRepository.kt

```diff
@@ -47,6 +48,83 @@ class MoviesRepository {
 
     @ExperimentalSerializationApi
     suspend fun loadRunTime(id: Long): Int = withContext((Dispatchers.IO)){
-        RemoteDataStore.getRuntime(id)
+        dataStore.getRuntime(id)
+    }
+
+    private fun convertMovieToMovieDB(movies: List<Movie>) : List<MoviesTableEntity> {
+        return movies.map { movie->
+            MoviesTableEntity(
+                    id = movie.id,
+                    originalTitle = movie.title,
+                    overview = movie.overview,
+                    posterPath = movie.poster,
+                    backdropPath = movie.backdrop,
+                    voteAverage = movie.ratings,
+                    voteCount = movie.numberOfRatings,
+                    adult = movie.minimumAge.toLong(),
+                    runTime = movie.runtime.toLong(),
+                    genreIDS = movie.genres.map { it.id }
+            )
+        }
+    }
+
+    private suspend fun convertMovieToMovieScreen(movies: List<MovieWithActors>): List<Movie>{
+        val genre1: List<Genre> = readGenreDB()
+        val genresMap1 = genre1.associateBy { it.id }
+        return movies.map { movie->
+            Movie(
+                    id = movie.moviesTableEntity.id,
+                    title = movie.moviesTableEntity.originalTitle,
+                    overview = movie.moviesTableEntity.overview,
+                    poster = movie.moviesTableEntity.posterPath,
+                    backdrop = movie.moviesTableEntity.backdropPath,
+                    ratings = movie.moviesTableEntity.voteAverage,
+                    numberOfRatings = movie.moviesTableEntity.voteCount,
+                    minimumAge = movie.moviesTableEntity.adult.toInt(),
+                    runtime = movie.moviesTableEntity.runTime.toInt(),
+                    genres = movie.moviesTableEntity.genreIDS.map { genresMap1[it] ?: throw IllegalArgumentException("Genre not found") },
+                    actors = movie.actors
+            )
+        }
+    }
+
+    private fun convertMovieToActorsDB(idMovie:Long, actors: List<Actor>) : List<ActorsTableEntity>{
+        return actors.map { movie->
+            ActorsTableEntity(
+                    id_Movie = idMovie,
+                    id = movie.id,
+                    name = movie.name,
+                    picture = movie.picture
+            )
+        }
+    }
+
+    suspend fun readMovieDB(id: Long) : Movie{
```

Comment:
@sanya-arsenal Not a good name for such a method. `getMovieForDB` sounds better.